### PR TITLE
[v8] Remove dependency on the Promises SDK

### LINKED ISSUE #107: Consider removing Promises dependency in major version update
The Promises dependency is currently broken on macOS because it hasn't yet updated from a 10.10 minimum version.

On closer look, the dependency seems heavyweight for a single public API that is used in just one place in Firebase.

In Firebase:
```
$ git grep gul_dataTaskPromiseWithRequest
FirebaseAppCheck/Sources/Core/APIService/FIRAppCheckAPIService.m:  return [self.URLSession gul_dataTaskPromiseWithRequest:request]
FirebaseAppCheck/Tests/Unit/Core/FIRAppCheckAPIServiceTests.m:  OCMExpect([URLSessionMock gul_dataTaskPromiseWithRequest:URLRequestValidationArg])
```

### PR DESCRIPTION
**Pull request by ncooke3**
Everything has been reviewed in a previous PR, minus the changelog entry introduced in this PR.

Will be followed by a re-tag and re-stage.

Fix #107

### CODE REVIEW

**Review by paulb777 — APPROVED** [review]
Nice!

File: GoogleUtilities/Environment/Public/GoogleUtilities/GULKeychainStorage.h [hunk 1]
```diff
@@ -34,42 +32,49 @@ NS_ASSUME_NONNULL_BEGIN
...
+/// @param accessGroup The Keychain Access Group.
+/// @param completionHandler The completion handler to call when the
+/// synchronized keychain removal is complete. An error is passed to the
+/// completion handler if the keychain removal fails. Else, `YES` if the item
+/// was removed successfully or doesn’t exist, `NO` otherwise.
+/// @note In the event an error occurs, the completion handler will return an
+/// error and a boolean to indicate the removal failed (`NO`).
+- (void)removeObjectForKey:(NSString *)key
+               accessGroup:(nullable NSString *)accessGroup
+         completionHandler:(void (^)(BOOL success, NSError *_Nullable error))completionHandler;
```
Comment (ncooke3) [inline GoogleUtilities/Environment/Public/GoogleUtilities/GULKeychainStorage.h:77]: @paulb777 @andrewheard It's minor, but I'm second guessing the non-nullable boolean parameter. I'm wondering if it's better to drop `BOOL success` and let the presence of an error determine if the remove was a success. Thoughts?

Comment (ncooke3) [reply GoogleUtilities/Environment/Public/GoogleUtilities/GULKeychainStorage.h:77]: My reasons against the `BOOL success` is:
- I don't like how it's non-null and the other param is nullable.
- Introduces a risk to bugs where success == NO but error is non-null

The only reasons I have for it are that is makes the API more explicit.

Comment (paulb777) [reply GoogleUtilities/Environment/Public/GoogleUtilities/GULKeychainStorage.h:77]: +1 to drop the BOOL argument if we're convince that there is a 1-1 correspondence between YES and nil error. and NO and non-nil error

Comment (andrewheard) [reply GoogleUtilities/Environment/Public/GoogleUtilities/GULKeychainStorage.h:77]: I agree with Paul. Quickly looking at the surface level in https://github.com/google/GoogleUtilities/pull/187/files#diff-97832c72a18629683286d1edd1d5d63ae846711843f24de5b0ba337339583879R112-R115 it looks possible for the error to be `nil` and success to be `NO`. Would need to dig deeper into `removeItemWithQuery` to see how it is implemented to know if that's actually a concern.

**Review by andrewheard — APPROVED** [review]

File: GoogleUtilities/Environment/Public/GoogleUtilities/GULKeychainStorage.h (hunk 1 above)
Comment (ncooke3) [reply GoogleUtilities/Environment/Public/GoogleUtilities/GULKeychainStorage.h:77]: Thanks y'all. See #188.

**Pull request merged**